src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateClone = { ...state };
+  for (const action of actions) {
+    const { type } = action;
```

Comment:
You need to destructure all properties
```suggestion
    const { type, extraData, keysToRemove } = action;
```